core/src/nucleus/ribosome/callback/validate_entry.rs

```diff
@@ -52,6 +52,10 @@ pub fn validate_entry(
             context,
         )?),
 
+        // Deletion entries are not validated currently and always valid
+        // TODO: Specify how Deletion can be commited to chain.
+        EntryType::Delete => Ok(CallbackResult::Pass),
```

Comment:
Deletion should be validated with the same validation callback used for commit. You are already setting the action field in the validation data to `EntryAction::Delete` above. That gets passed into the callback anyway.

Reply:
I don't understand how it can be same validation as commit/create we are not putting new data that needs to be validated.

Reply:
From a DNA developer perspective: the same (the one and only) validation callback is called, but with a `EntryAction::Delete`. The content should be the existing (to be deleted) entry.